src/crush/CrushLocation.cc

```diff
@@ -43,14 +45,120 @@ int CrushLocation::_parse(const std::string& s)
   return 0;
 }
 
+#if defined(WITH_SEASTAR) && !defined(WITH_ALIEN)
+namespace {
+  seastar::logger& logger() {
+    return crimson::get_logger(ceph_subsys_crush);
+  }
+}
+
+using namespace crimson::common;
+seastar::future<> CrushLocation::update_from_hook()
+{
+  if (local_conf().get_val<std::string>("crush_location_hook").length() == 0)
+    return seastar::now();
+
+  return seastar::file_exists(local_conf().get_val<std::string>("crush_location_hook")).then([this] (bool result) {
```

Comment:
Formatting, please split this line (and the other long ones).
https://google.github.io/styleguide/cppguide.html#Line_Length

Reply:
done